sys/include/suit/transport/worker.h

```diff
@@ -39,6 +39,48 @@ extern "C" {
  */
 void suit_worker_trigger(const char *url, size_t len);
 
+/**
+ * @brief   Trigger a SUIT udate via a worker thread
+ *
+ * @pre The caller called @ref suit_worker_prepare to obtain the @p buf, and
+ * populated @p size bytes into it.
+ *
+ * This can be called with a size of 0 when writing a manifest was started, but
+ * could not be completed.
+ *
+ * @param[in] manifest  Pointer to the manifest. This must be the return value
+ *                      of a previous call to @ref suit_worker_prepare, and is
+ *                      invalidated at some point during or after the call.
+ * @param[in] size      Number of bytes that have been prepared in @ref
+ *                      the buffer before the call.
+ */
+void suit_worker_trigger_prepared(const uint8_t *manifest, size_t size);
+
+/**
+ * @brief    Prepare for a worker run with a preloaded manifest
+ *
+ * This obtains a lock on the SUIT worker, and provides a pointer to a memory
+ * area into which the manifest is to be written. The lock must be released by
+ * calling @ref suit_worker_trigger_prepared later.
+ *
+ * @param[out]   buffer    On success, buffer into which the image may be
+ *                         written.
+ * @param[inout] size      Requested buffer size. On some errors, this will be
+ *                         decreased to a size that would be acceptable.
+ *
+ * @return 0 on success
+ * @return -EAGAIN if the worker is currently busy.
```

Comment:
I'm not a huge fan of an API that requires polling on the caller side.
Do you have a use-case in mind that requires this to be non-blocking?

Reply:
That's the nature of anything related to the worker thread already -- and I think that's fine because:

a) We can't make this blocking without major work, because one can't block on a thread to terminate, only on the mutex it is holding, but even if that mutex dropping is the last thing that thread does, there is a possible race condition depending on the thread priorities. (It's the whole thread return value discussion).
b) When triggered from a CoAP server, blocking would block the whole CoAP thread, racking up retransmits and generally not serving the peers while this is being processed.

And most importantly, if this is blocked, it's because there is an active manifest transfer in process. Chances are the user will not even *want* this to block, but rather to see that something is afoot, and to come back after having reviewed the whole situation.

Reply:
Or, from a different PoV: This is an allocation-ish situation (just that it doesn't happen from the general heap but from the 1-sized pool of possible transfers). A malloc doesn't block either.

Reply:
Ok in that context it makes sense - and if the user is able to block, they can just call `suit_handle_prepared()` themselves.